Python/sampleclient/filechecks.py

```diff
@@ -0,0 +1,40 @@
+import os
+from pathlib import Path
+
+
+def file_exists(file_path):
+    '''Returns if file exists or not.'''
+    file_path = Path(file_path)
+    return file_path.exists()
+
+
+def file_size_in_bytes(file_path):
+    '''Return size of file in bytes, or -1 if file does not exist.'''
+    if not file_exists(file_path):
+        return None
+    statinfo = os.stat(file_path)
+    return int(statinfo.st_size)
+
+
+MAX_FILE_SIZE_FOR_PUT_API = int(1024 * 1024 * 4)
+
+
+def files_must_not_be_more_than(files, file_size_limit_in_bytes):
```

Comment:
Following up on this earlier thread: https://github.com/Microsoft/msrd-rest-samples/pull/22#discussion_r275583981

I think the "teaching" comment is exactly the right idea for this script.

I still think this is presenting new idioms and lines of code without enough payoff for the user. I'd like, where appropriate, to make assumptions about function inputs, and use those assumptions to simplify the code, especially when a violation of those assumptions means we actually want to terminate the script.

All the code in this module could be replaced by the following at its sole use site:
```python
try:
    total_size = sum(Path(f).stat().st_size() for f in files)

    if total_size > MAX_SIZE:
        excess = total_size - MAX_SIZE

        error_out(  # ...log error about excess
except OSError as e:
    # log that file wasn't found, exit with message
    error_out(  #...
```
You could then use `Path(some_filename).exists()` wherever you currently use `file_exists()`.

If you want to encapsulate some logic in functions, how about we handle each of the caller `storage_type` cases in a method, and dispatch to the handler method in `FileUpload::upload_files()`?

Reply:
Thanks for your patience, I was out sick with a fever for a few days.

May I ask what the standard you want kept to for keeping code DRY is, then?  I was trying to be verbose and explicit, but also DRY, as part of attempting to add some SOLID practices to the code in order to keep it open for extension yet still keep the code closed for modification and improve testability.

If I understand what you are saying here, you would rather just the code "error out and die" if thats the only/last functional resort; And I get that functionally, for a developer user, that makes sense if we don't care about unit tests or testability.

But what about unit testability?  Unit tests are why I was pulling the new error_out(...) method out of that code, so it was a separate concern that didn't need to be worked around since if exit() gets called in a unit test, they fail.

But also going back to my teaching comment, I'm not sure I can assume in a safe way that the user using this is a developer, or that they have any technical skills; So I went out and asked Support and as a result, I am not sure I can assume the user knows how to look at a stacktrace and know what to do with it, and I was working at this from a user accessibility perspective from the angle of "If it can confuse them, it will generate support tickets, so how do I keep support tickets as low as possible?".  So I concluded that keeping the code explicit (PEP20) was going to better serve our interests

Did you want me to deprioritize unit tests more?

Reply:
You can assume the user of this script is a developer or pen tester.